### PR TITLE
Feat/non_app_ota_update

### PR DESCRIPTION
**Pull request by hrushikesh430**
# Checklist

-  Component contains License
-  Component contains README.md
-  Component contains idf_component.yml file with `url` field defined
- [ ] Add component example
- [ ] Add unit tests for component
- [ ] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [ ] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Change description
- Component to do OTA for non-app partition. This PR is continuation of this [PR](https://github.com/espressif/idf-extra-components/pull/328).

### CODE REVIEW

File: esp_custom_part_ota/include/esp_custom_part_ota.h [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * @return
+ *      - ESP_OK
+ *      - ESP_ERR_INVALID_ARG
+*/
+esp_err_t esp_custom_part_ota_end(esp_custom_part_ota_handle_t handle);
+
+
+/**
+ * @brief   Abort the OTA update process and free the handle
+ *
```
Comment (AdityaHPatwardhan) [inline esp_custom_part_ota/include/esp_custom_part_ota.h:70]: We can mention here that After aborting the OTA, you need to call `... restore_ota ...` to restore the backup if available

File: esp_custom_part_ota/include/esp_custom_part_ota.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ *      - ESP_ERR_INVALID_ARG
+ *      - lower level esp_partition error codes.
+*/
+esp_err_t esp_custom_part_ota_write(esp_custom_part_ota_handle_t handle, const void *data, size_t size);
+
+
+/**
+ * @brief   Finish OTA update
+ *
+ * @param ctx   esp_custom_part_ota_handle_t
```
Comment (AdityaHPatwardhan) [inline esp_custom_part_ota/include/esp_custom_part_ota.h:59]: This API invalidates the backup data if available.
Please note that this does not erase the actual data on the flash. For that you may need to explicitly erase the backup partition

File: esp_custom_part_ota/src/esp_custom_part_ota.c [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+esp_err_t esp_custom_part_ota_abort(esp_custom_part_ota_handle_t handle)
+{
+    if (!handle) {
+        return ESP_ERR_INVALID_ARG;
+    }
+    esp_custom_part_ota_t *ctx = (esp_custom_part_ota_t *)handle;
+    free(ctx);
+    return ESP_OK;
```
Comment (AdityaHPatwardhan) [inline esp_custom_part_ota/src/esp_custom_part_ota.c:113]: Set the NVS entry to  0

**Comment by mahavirj** [comment]
Closing as the IDF `app_update` component has now been extended to support custom partition OTA as well.

**Pull request closed without merging**